float32 voltage
uint8 maximum_level
uint8 current_level
bool charging